limit-pull-requests/action.yml

```diff
@@ -0,0 +1,82 @@
+name: Limit pull requests
+description: Limit the number of pull requests to the repository created by a user
+author: ZhongRuoyu
+branding:
+  icon: alert-triangle
+  color: yellow
+
+inputs:
+  token:
+    description: GitHub token
+    required: false
+    default: ${{ github.token }}
+  limit:
+    description: The maximum number of pull requests allowed
+    required: true
+    default: "10"
+  except-users:
+    description: The users exempted from the limit, one per line
+    required: false
+  # https://docs.github.com/en/graphql/reference/enums#commentauthorassociation
+  except-author-associations:
+    description: The author associations exempted from the limit, one per line
+    required: false
+  comment:
+    description: The comment to post when the limit is reached
+    required: false
+  close:
+    description: Whether to close the pull request when the limit is reached
+    required: true
+    default: "false"
+
+runs:
+  using: composite
+  steps:
+    - name: Check the number of pull requests
+      id: count-pull-requests
+      run: |
+        # If the user is exempted, assume they have no pull requests.
+        if grep -Fqx '${{ github.actor }}' <<<'${{ inputs.except-users }}'; then
+          echo "::notice::@${{ github.actor }} is exempted from the limit."
+          echo "count=0" >>"$GITHUB_OUTPUT"
+          exit 0
+        fi
+        if grep -Fqx '${{ github.event.pull_request.author_association }}' <<<'${{ inputs.except-author-associations }}'; then
+          echo "::notice::@{{ github.actor }} is a ${{ github.event.pull_request.author_association }} exempted from the limit."
+          echo "count=0" >>"$GITHUB_OUTPUT"
+          exit 0
+        fi
+
+        count="$(
+          gh pr list \
+            --state=open \
+            --author='${{ github.actor }}' \
```

Comment:
Is this right? Looks like it'll filter by whichever user is running this action?

Reply:
I think it's correct if we invoke this action in a `pull_request_target` event. We also have this in the triage workflow in `homebrew-core`, where I think this action will be added:

```yaml
      - name: Check commit format
        if: >
          github.actor != 'BrewTestBot' &&
          !contains(github.event.pull_request.labels.*.name, 'CI-published-bottle-commits')
```

https://github.com/Homebrew/homebrew-core/blob/21f41eca9c7fab6d75b84906f15847b226d5ff0f/.github/workflows/triage.yml#L63-L66

Reply:
Ok, I see now, thanks!